packages/server/src/automations/loopUtils.ts

```diff
@@ -0,0 +1,136 @@
+import { AutomationErrors } from "../constants"
+import * as automationUtils from "./automationUtils"
+import env from "../environment"
+import { LoopInput } from "../definitions/automations"
+
+export function initializeTempOutput(loopSteps: any, iterationCount: number) {
+  return { items: loopSteps, iterations: iterationCount }
+}
+
+export function typecastLoopInput(
```

Comment:
I don't know if the name of this is quite right - or it should probably return the typecast loop input rather than operating on an in/out parameter mechanism. I assume the boolean returned is to state whether or not it was able to update typecast, but I think a lot of logic could be extracted from this into the caller if all this did was try to typecast and if it fails throw an error. Then the caller could manage setting the incorrect type and whether to proceed.